tests/src/test/scala/whisk/core/containerpool/logging/SplunkLogStoreTests.scala

```diff
@@ -0,0 +1,207 @@
+import whisk.core.entity.LogLimit
+import whisk.core.entity.MemoryLimit
+import whisk.core.entity.Parameters
+import whisk.core.entity.Subject
+import whisk.core.entity.TimeLimit
+import whisk.core.entity.WhiskActivation
+import whisk.core.entity.size._
+
+object SplunkLogStoreTests {
+  val config = """
+    whisk.logstore {
+        log-driver-message = "Logs are stored externally."
+        log-driver-opts = [
+            { "--log-driver" : "fluentd" },
+            { "--log-opt" : "tag=OW_CONTAINER" },
+            { "--log-opt" : "fluentd-address=localhost:24225" }
+        ]
+        splunk {
+            host = "splunk-host"
+            port = 8080
+            user = "splunk-user"
+            password = "splunk-pass"
+            index = "splunk-index"
+            log-message-field = "log_message"
+            activation-id-field = "activation_id"
+            disableSNI = false
+        }
+    }
+    """
+
+}
+
+class SplunkLogStoreTests
+    extends TestKit(ActorSystem("SplunkLogStore", ConfigFactory.parseString(SplunkLogStoreTests.config)))
+    with FlatSpecLike
+    with Matchers
+    with ScalaFutures {
+  behavior of "Splunk LogStore"
+
+  val startTime = "2007-12-03T10:15:30"
+  val endTime = "2007-12-03T10:15:45"
+
+  val activation = WhiskActivation(
+    namespace = EntityPath("ns"),
+    name = EntityName("a"),
+    Subject(),
+    activationId = ActivationId(),
+    start = LocalDateTime.parse(startTime).toInstant(ZoneOffset.UTC),
+    end = LocalDateTime.parse(endTime).toInstant(ZoneOffset.UTC),
+    response = ActivationResponse.success(Some(JsObject("res" -> JsNumber(1)))),
+    annotations = Parameters("limits", ActionLimits(TimeLimit(1.second), MemoryLimit(128.MB), LogLimit(1.MB)).toJson),
+    duration = Some(123))
+
+  implicit val ec = system.dispatcher
+  implicit val materializer = ActorMaterializer()
+
+  val splunkIndex = system.settings.config.getString("whisk.logstore.splunk.index")
+  val activationIdFieldName = system.settings.config.getString("whisk.logstore.splunk.activation-id-field")
+  val logMessageFieldName = system.settings.config.getString("whisk.logstore.splunk.log-message-field")
+  val splunkUser = system.settings.config.getString("whisk.logstore.splunk.user")
+  val splunkPwd = system.settings.config.getString("whisk.logstore.splunk.password")
+  val splunkHost = system.settings.config.getString("whisk.logstore.splunk.host")
+  val splunkPort = system.settings.config.getInt("whisk.logstore.splunk.port")
+
+  val testFlow: Flow[(HttpRequest, Promise[HttpResponse]), (Try[HttpResponse], Promise[HttpResponse]), NotUsed] =
+    Flow[(HttpRequest, Promise[HttpResponse])]
+      .mapAsyncUnordered(1) {
+        case (request, userContext) =>
+          //handle assertion errors so that stream does not swallow error info
+          try {
+            request.uri.path.toString() shouldBe "/services/search/jobs"
+            request.headers shouldBe List(Authorization.basic(splunkUser, splunkPwd))
```

Comment:
Can these move into the inner try block to get away some nesting?

Reply:
yes